lib/Dialect/ESI/ESIOps.cpp

```diff
@@ -72,6 +72,22 @@ LogicalResult ChannelBufferOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// FIFO functions.
+//===----------------------------------------------------------------------===//
+
+LogicalResult FIFOOp::verify() {
+  ChannelType inputType = getInput().getType();
+  if (inputType.getSignaling() != ChannelSignaling::FIFO)
+    return emitOpError("only supports FIFO signaling on input");
+  ChannelType outputType = getOutput().getType();
+  if (outputType.getSignaling() != ChannelSignaling::FIFO)
+    return emitOpError("only supports FIFO signaling on output");
+  if (outputType.getInner() != inputType.getInner())
+    return emitOpError("input and output types must match");
+  return success();
```

Comment:
Missing verifiers for these

Reply:
You mean verifier tests?

Reply:
Yep!

Reply:
I'm only doing the inner type check test since the first two restrictions are being removed shortly.